lib/primitives/keyring.js

```diff
@@ -154,19 +155,26 @@ class KeyRing {
    * Inject data from public key.
    * @private
    * @param {Buffer} key
+   * @param {Boolean?} compress
+   * @returns {KeyRing}
    */
 
-  fromPublic(key) {
+  fromPublic(key, compress) {
     assert(Buffer.isBuffer(key), 'Public key must be a buffer.');
     assert(secp256k1.publicKeyVerify(key), 'Not a valid public key.');
-    this.publicKey = key;
+
+    if (typeof compress === 'boolean')
+      this.publicKey = secp256k1.publicKeyConvert(key, compress !== false);
+    else
+      this.publicKey = key;
+
     return this;
   }
 
   /**
    * Generate a keyring.
    * @private
-   * @param {Boolean?} compress
+   * @param {Boolean?} compress (default = true)
```

Comment:
Nit: our codebases use `[` and `]` for default values over `(` and `)`
See: https://github.com/bcoin-org/bledger/blob/master/lib/bcoin.js#L127